api/src/wfl/module/covid.clj

```diff
@@ -156,20 +158,131 @@
   (-> workload
       (util/select-non-nil-keys workload-metadata-keys)
       (dissoc :pipeline)
-      (update :source   util/to-edn)
+      (update :source util/to-edn)
       (update :executor util/to-edn)
-      (update :sink     util/to-edn)))
-
-(defoverload workloads/create-workload!    pipeline create-covid-workload)
-(defoverload workloads/start-workload!     pipeline start-covid-workload)
-(defoverload workloads/update-workload!    pipeline update-covid-workload)
-(defoverload workloads/stop-workload!      pipeline stop-covid-workload)
-(defoverload workloads/retry               pipeline batch/retry-unsupported)
-(defoverload workloads/load-workload-impl  pipeline load-covid-workload-impl)
-(defmethod   workloads/workflows           pipeline
+      (update :sink util/to-edn)))
+
+(defn retry-workflows
+  [workflow workloads]
+
+  ; 1. load the workload
+
+
+  ; 1. Get the workflows of status X and null retry field
+  ;(let [workflows (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+  ;                                          (executor/executor-workflows-by-status tx {:executor workload} workflow status))]
+  ;  (println workflows))
+
+  ; 2. Get the distinct list of snapshots for these workflows
+
+  ; 3. Submit the snapshots. Hold onto the new workflow ids
+
+  ; 4. Update the retry field of the executor details table inserting the new workflow id for the old workflow.
+
+)
+
+(comment
+  (let [tx (jdbc/with-db-transaction [tx (postgres/wfl-db-config)])
+        w (workloads/load-workload-for-uuid tx "b8b5bbe6-faad-4d03-ad45-a68f6c860c7b")]
+    (pprint w)
+    (pprint (util/to-edn w)))
```

Comment:
You cannot return `tx` from this scope - it's lifetime is bound to the body in which it's created
```clojure
(jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
  (let [w (workloads/load-workload-for-uuid tx "b8b5bbe6-faad-4d03-ad45-a68f6c860c7b")]
    (pprint w)
    (pprint (util/to-edn w))))
```

Reply:
I tried this:
`(jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
    (let [w (workloads/load-workload-for-uuid tx "b8b5bbe6-faad-4d03-ad45-a68f6c860c7b")]
      (pprint w)
      (pprint (util/to-edn w))))`

Same result - 
`04:14:54 INFO  wfl.module.covid - JDBC transaction 2495 started to "ranthony@jdbc:postgresql:wfl#9806"
Execution error (NoSuchMethodError) at clojure.java.jdbc/db-transaction* (jdbc.clj:838).
clojure.java.jdbc$db_transaction_STAR_$fn__8398.<init>(ZLjava/lang/Object;)V`